### DISCUSSION TITLE
How to make docker work?

### DISCUSSION THREAD

**Discussion by adocampo** [euzu/tuliprox#54] [Q&A] [answered]
I'm not new to docker, but I'm unable to make m3u-filter even start witht the docker-compose.yml on the repository or with the built image, both says
```
ERROR m3u_filter] M3uFilter error: cant read config file: sources[0].targets[0].type: invalid type: string "m3u", expected a sequence at line 65 column 15
```
What could be happening?

**Comment by euzu** [comment]
@adocampo  the docker image has an example configuration  which is not meant to be used, it has only examples and are not  valid yaml files.
The docker compose file mounts relative directories. One of the directory is the config directory. Please look at the wiki or read the readme to create a individual configuration.

**Comment by euzu** [comment, marked as answer]
@adocampo here is a description for a xtream provider.

[xtream](https://github.com/euzu/m3u-filter/wiki/scenario_xtream_provider)

**Reply by adocampo** [reply]
The URL is something like 
http://domain.example.com:80/get.php?username=ad123456&password=12345678type=m3u_plus&output=mpegts

I gues I could split into URL, username and password, as I think it would work with xtream

**Comment by euzu** [comment]
@adocampo  this depends on your provider.
what you have is a m3u playlist link which is supported by m3u-filter. But if you want to use the xtream api and your provider supports it
then the url is the part from htt...:80  in your example. The username and password are  ad123456 and 12345678.

m3u-filter can create xtream output from m3u url but this is not recommended, because you loose the series.
